web/src/main/java/org/akaza/openclinica/controller/InsightReportController.java

```diff
@@ -0,0 +1,208 @@
+                    }
+                }
+            }
+        }
+
+        return new ResponseEntity(HttpStatus.OK);
+    }
+
+    public String[] getLabels(String label) {
+        // original label/id not contain replicaSubString
+        // replica label/id contain replicaSubString
+        String[] labels = new String[2];
+        if (label.indexOf(replicaSubstring) > -1) {
+            labels[0] = label.split(replicaSubstring)[0];
+            labels[1] = label;
+        } else {
+            labels[0] = label;
+            labels[1] = label + replicaSubstring;
+        }
+        return labels;
+    }
+
+    public String doPost(String api, HashMap<String, String> mapHeaders,
+                         String body, HashMap<String, String> queries) {
+        try {
+            HttpHeaders headers = new HttpHeaders();
+            for (String i : mapHeaders.keySet()) {
+                headers.add(i, mapHeaders.get(i).toString());
+            }
+
+            HttpEntity<String> request;
+            if (body != null) {
+                request = new HttpEntity<>(body, headers);
+            } else {
+                request = new HttpEntity<String>(headers);
+            }
+            RestTemplate rest = new RestTemplate();
+            ResponseEntity<String> response;
+            if (queries != null) {
+                response = rest.postForEntity(api, request, String.class, queries);
+            } else {
+                response = rest.postForEntity(api, request, String.class);
+            }
+            return response.getBody();
+        } catch (Exception e) {
+            System.out.println("## Error :: "+e.getMessage());
+            return null;
+        }
+
+    }
+
+    public String getFilePath(String label) {
+        return userService.getFilePath(JobType.INSIGHT_REPORT) + File.separator + label + UNDERSCORE + reportName +
+                new SimpleDateFormat("_yyyy-MM-dd-hhmmssS'.txt'").format(new Date());
+    }
+
+    public String saveToFile(String content, String label) {
+        FileOutputStream fop = null;
+        File file;
+        String fileName = null;
+        try {
+
+            file = new File(getFilePath(label));
+            fop = new FileOutputStream(file);
+
+            // if file doesnt exists, then create it
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+
+            fileName = file.getName();
+
+            // get the content in bytes
+            byte[] contentInBytes = content.getBytes();
+
+            fop.write(contentInBytes);
+            fop.flush();
+            fop.close();
+
+        } catch (IOException e) {
+            System.out.println("=== Error create file : " + e.getMessage());
+            e.printStackTrace();
+        } finally {
+            try {
+                if (fop != null) {
+                    fop.close();
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
Please log the exception. Do not print it.
Log it like this:
logger.error("Some message..." + e);

Reply:
code updated